src/Http/Http/src/BindingAddress.cs

```diff
@@ -217,6 +217,8 @@ public static BindingAddress Parse(string address)
                     hasSpecifiedPort = true;
                     host = address.Substring(schemeDelimiterEnd, portDelimiterStart - schemeDelimiterEnd);
                     port = portNumber;
+                }else{
+                    throw new FormatException($"Invalid port: '{portString}'");
```

Comment:
Is `FormatException` correct? In https://github.com/dotnet/aspnetcore/issues/24384#issuecomment-1261248502 it's just "more specific".

Maybe there should also be a "help-message" like

```suggestion
                    throw new FormatException($"Invalid port: '{portString}, only numbers are allowed'");
```
Although that should be obvious for ports...